src/ic_app.cpp

```diff
@@ -27,6 +30,40 @@ namespace IC {
         EventDispatcher dispatcher(e);
         dispatcher.Dispatch<WindowCloseEvent>(BIND_EVENT_FN(OnWindowClose));
         dispatcher.Dispatch<WindowResizeEvent>(BIND_EVENT_FN(OnWindowResize));
+        dispatcher.Dispatch<KeyPressedEvent>(BIND_EVENT_FN(OnKeyPressed));
+
+        // Propagate events up the layer stack and cancel the
+        // event propagation if the event is marked as "handled".
+        for (auto it = _layerStack.end(); it != _layerStack.begin();) {
```

Comment:
There's also `rbegin` and `rend` functions for a vector that return "reverse iterators" so you don't have to iterate using predecrement -- instead of ++.

Reply:
Fair, though I think here, I want to be explicit that we're going backwards through the stack. The `end()` and `begin()` members here are also being exposed from the class, this isn't direct access to the vector.